can/io/player.py

```diff
@@ -88,6 +92,25 @@ def __new__(  # type: ignore
                 f'No read support for this unknown log format "{suffix}"'
             ) from None
 
+    @staticmethod
+    def decompress(
+        filename: "can.typechecking.StringPathLike",
+    ) -> typing.Tuple[str, typing.IO[typing.Any]]:
+        """
+        Return the suffix and io object of the decompressed file.
+        """
+        real_suffix = pathlib.Path(filename).suffixes[-2].lower()
+        file = gzip.open(filename, "rt")
+
+        try:
+            for line in file:
+                str(line)
+            file.seek(0)
+        except UnicodeDecodeError:
+            file = gzip.open(filename, "rb")
```

Comment:
Isn't this required?

```suggestion
            file.close()  # Close the text file pointer
            file = gzip.open(filename, "rb")
```

Reply:
Not sure it's _required_, but definitely a good idea to not rely on the gc.